Samples/DirectMLNpuInference/main.cpp

```diff
@@ -0,0 +1,183 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+
+#include <dxcore_interface.h>
+#include <dxcore.h>
+
+#include "onnxruntime_cxx_api.h"
+#include "dml_provider_factory.h"
+
+#include "TensorHelper.h"
+
+using Microsoft::WRL::ComPtr;
+
+void InitializeDirectML(ID3D12Device1** d3dDeviceOut, ID3D12CommandQueue** commandQueueOut, IDMLDevice** dmlDeviceOut) {
+    // Whether to skip adapters which support Graphics in order to target NPU for testing
+    bool forceComputeOnlyDevice = true;
+    ComPtr<IDXCoreAdapterFactory> factory;
+    HMODULE dxCoreModule = LoadLibraryW(L"DXCore.dll");
```

Comment:
Should we add comment on why we are dynamically loading the dlls instead of calling exports from dll directly

Reply:
There is no significant reason, but this was the pattern previously shared in the devkit documentation.